util/reggen/gen_tock.py

```diff
@@ -0,0 +1,384 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Generate Rust constants from validated register JSON tree
```

Comment:
Maybe its is a good idea to describe how this differs from the `gen_rust.py` module? I presume they generate a different files, since it seems this was copied and modified?

Reply:
or is the thought that the `gen_rust.py` module will be removed in favor of this one? (if thats the case, the comments about code duplication are irrelevant).